pyp2spec/pyp2conf.py

```diff
@@ -44,14 +45,15 @@ class PypiPackage:
     This is only relevant for testing.
     """
 
-    def __init__(self, package_name, *, version=None, package_metadata=None, session=None):
+    def __init__(self, package_name, *, version=None, package_metadata=None, metadata_from_file=None, session=None):
         self.package_name = package_name
         self._session = session or requests.Session()
         self.version = version or self._get_version_from_package_metadata()
         # package_metadata - custom dictionary with package metadata - used for testing
         # in the typical app run it's not set,
         # meaning we jump to the other sources package metadata data (eg. PyPI)
         self.package_data = package_metadata or self._get_package_version_metadata()
+        self.metadata_from_file = metadata_from_file or self.parse_metadata()
```

Comment:
I don't understand why those two are separate dicts.
Would it make sense to just keep one?

```suggestion
        self.package_data = package_metadata or self._get_package_version_metadata()
        self.package_data.update(self.parse_metadata())
```

or even

```suggestion
        self.package_data = package_metadata or self._get_package_version_metadata()
        self.parse_metadata()  # updates self.package_data in-place
```

Reply:
The con: It's been something over a year since PyPI serves the metadata file with the wheels. Many projects will not have it yet.

Reply:
OK. Makes sense. It's just that the `package_metadata, metadata_from_file` init parameters are a bit confusing/convoluted :/

Reply:
Let me try to disambiguate the naming.

Reply:
Now the naming should be better: `pypi_package_data` - read from PyPI API, `core_metadata` - read from the metadata file